packages/core/index.d.ts

```diff
@@ -115,6 +115,7 @@ declare module '@rjsf/core' {
         multiple: boolean;
         rawErrors: string[];
         registry: Registry;
+        type?: string; // Add the optional prop for TextWidget to simplify Typescript usage
```

Comment:
Yes, only `TextWidget` will use `type`

Reply:
Why? (I'm not too familiar with that code and I'm guessing you've already looked deeply into this.)

Reply:
If you look at the `bootstrap-4` code you'll see that it passes the `type` down to `TextField`. The custom `TextFieldProps` didn't "cast" properly from a `WidgetProp` due to the optional `type` field. I tried doing something similar to what `material-ui` does, but bootstrap didn't like it. It seemed simpler just to add an optional `type` to the `WidgetProps` to simplify the conversion. Since it is optional it also happens to be completely backwards compatible in Typescript.

Reply:
@epicfaace Any another concerns?

Reply:
@epicfaace I took a page from the `FieldProps` book and added a generic "other props" option that also works